src/modules/.internal/state/__tests__/index.test.ts

```diff
@@ -0,0 +1,47 @@
+import { I18nStateManager } from '../';
+import { I18nState } from '../types';
+
+describe('I18nStateManager', () => {
+  beforeEach(() => {
+    I18nStateManager.resetInstance();
+  });
+
+  it('should return the same instance when calling getInstance', () => {
+    const instance1 = I18nStateManager.getInstance();
+    const instance2 = I18nStateManager.getInstance();
+
+    expect(instance1).toBe(instance2);
+  });
+
+  it('should have default state when getting the state', () => {
+    const instance = I18nStateManager.getInstance();
+    const defaultState = {
+      locale: 'en-IN',
+      direction: 'ltr',
```

Comment:
What is direction here ?

Reply:
direction in which text is written, like it varies for Urdu and English.
Have added this just as default, it's not being used anywhere.